File: iSeconds.Domain/Timeline.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using SQLite;

namespace iSeconds.Domain
{
	public class Timeline : IModel
	{
		private IRepository repository;

	   public Timeline(string name, int userId)
		{
			Name = name;
			UserId = userId;
		}

		public Timeline()
		{
		}

		public void SetRepository(IRepository repository)
		{
			this.repository = repository;
		}

		public void AddVideoAt(DateTime date, string url)
		{
			Debug.Assert(repository != null);

			DayInfo day = repository.GetDayInfoAt(date, Id);
			day.AddVideo(url);
		}

		public void DeleteVideoAt (DateTime date, string url)
		{
			Debug.Assert (repository != null);

			DayInfo day = repository.GetDayInfoAt(date, Id);
			day.DeleteVideo (url);
		}

		public void DeleteAllVideos()
		{
			Debug.Assert (repository != null);

			var days= repository.GetAllDays(this.Id);
			foreach (DayInfo day in days)
				day.DeleteVideos();
		}

		public IList<MediaInfo> GetVideosAt(DateTime date)
		{
			Debug.Assert(repository != null);

			DayInfo day = repository.GetDayInfoAt(date, Id);
			return day.GetVideos();
		}

		public DayInfo GetDayAt(DateTime date)
		{
			Debug.Assert(repository != null);

			return repository.GetDayInfoAt(date, Id);
		}

		#region db

		public int UserId { get; set; }

		public string Name { get; set; }

	   public string Description { get; set; }

	   [PrimaryKey, AutoIncrement]
		public int Id { get; set; }

	   #endregion
	}
}

File: jYester/src/iSeconds/Domain/Timeline.java
package iSeconds.Domain;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import com.activeandroid.Model;
import com.activeandroid.annotation.Column;
import com.activeandroid.annotation.Table;

@Table(name = "Timeline")
public class Timeline extends Model {
	
	private IRepository repository = null;
	
	public Timeline(String name, int userId) {
		this.name = name;
		this.userId = userId;
	}
	public Timeline() {}
	
	public void setRepository(IRepository repository) {
		this.repository = repository;
	}
	
	public void addVideoAt(Date date, String url) {
		
		assert(repository != null);

		DayInfo day = repository.getDayInfoAt(date, this.getId());
		day.addVideo(url);
	}
    
	public void deleteAllVideos() {
		
	} 
	
	public List<MediaInfo> getVideosAt(Date date) {
		assert(repository != null);

		DayInfo day = repository.getDayInfoAt(date, this.getId());
		return day.getVideos();
	}
	
	public List<DayInfo> getDays() {
		return repository.getAllDays(this.getId());
	}

//	public List<MediaInfo> getAllVideos() {
//		assert(repository != null);
//		
//		List<MediaInfo> medias= new ArrayList<MediaInfo>();
//		List<DayInfo> days= repository.getDays(this.getId());
//		for (DayInfo day: days) {
//			medias.addAll(day.getVideos());
//		}
//		
//		return medias;
//	}
	
	public List<DayInfo> getAllVideos() {
		assert(repository != null);		
		return repository.getDays(this.getId());
	}
	
	@Column(name = "Name")
	public String name;
	
	@Column(name = "UserId")
	public int userId;

	@Column(name = "Description")
	public String description;


}
